src/Minimap/Minimap.js

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback, useMemo } from 'react';
+import { useRef } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+import Nodes from './Nodes';
+import Viewport from './Viewport';
+
+const MINIMAP_WIDTH = 200;
+const MINIMAP_HEIGHT = 150;
+const MINIMAP_STYLE = {
+  boxSizing: 'content-box',
+  width: `${MINIMAP_WIDTH}px`,
+  height: `${MINIMAP_HEIGHT}px`,
+  backgroundColor: 'rgba(240, 240, 240, 1)',
+  position: 'fixed',
+  bottom: '20px',
+  left: '20px',
+};
+
+const Minimap = ({ vertices, extremeX, extremeY, viewport, changeScrollHandler }) => {
+  const scalingFactor = useMemo(
+    () => ({
+      x: MINIMAP_WIDTH / extremeX,
+      y: MINIMAP_HEIGHT / extremeY,
+    }),
+    [extremeX, extremeY]
+  );
+
+  const ref = useRef(null);
+  const element = ref.current;
+
+  const handleClick = useCallback(
+    event => {
+      const newScrollPosition = {
+        scrollLeft: (event.clientX - element.offsetLeft - (viewport.width * scalingFactor.x) / 2) / scalingFactor.x,
+        scrollTop: (event.clientY - element.offsetTop - (viewport.height * scalingFactor.y) / 2) / scalingFactor.y,
+      };
+      changeScrollHandler(newScrollPosition);
+    },
+    [element, viewport, scalingFactor]
```

Comment:
remove element from dependency and use directly `containerRef.current` inside this function.